server/src/test/java/org/elasticsearch/index/mapper/RangeFieldTypeTests.java

```diff
@@ -84,20 +87,50 @@ public void testRangeQuery() throws Exception {
         IndexSettings idxSettings = IndexSettingsModule.newIndexSettings(randomAlphaOfLengthBetween(1, 10), indexSettings);
         QueryShardContext context = new QueryShardContext(0, idxSettings, null, null, null, null, null, xContentRegistry(),
             writableRegistry(), null, null, () -> nowInMillis, null);
-        RangeFieldMapper.RangeFieldType ft = new RangeFieldMapper.RangeFieldType(type, Version.CURRENT);
+        RangeFieldType ft = new RangeFieldType(type, Version.CURRENT);
         ft.setName(FIELDNAME);
         ft.setIndexOptions(IndexOptions.DOCS);
 
-        ShapeRelation relation = RandomPicks.randomFrom(random(), ShapeRelation.values());
-        boolean includeLower = random().nextBoolean();
-        boolean includeUpper = random().nextBoolean();
+        ShapeRelation relation = randomFrom(ShapeRelation.values());
+        boolean includeLower = randomBoolean();
+        boolean includeUpper = randomBoolean();
         Object from = nextFrom();
         Object to = nextTo(from);
 
         assertEquals(getExpectedRangeQuery(relation, from, to, includeLower, includeUpper),
             ft.rangeQuery(from, to, includeLower, includeUpper, relation, null, null, context));
     }
 
+    public void testRangeQueryUseDateFormatInMapping() throws Exception {
+        Settings indexSettings = Settings.builder()
+            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).build();
+        IndexSettings idxSettings = IndexSettingsModule.newIndexSettings("test", indexSettings);
+        QueryShardContext context = new QueryShardContext(0, idxSettings, null, null, null, null, null, xContentRegistry(),
+            writableRegistry(), null, null, () -> nowInMillis, null);
+
+        RangeFieldType fieldType = new RangeFieldType(RangeType.DATE, Version.CURRENT);
+        fieldType.setName(FIELDNAME);
+        fieldType.setIndexOptions(IndexOptions.DOCS);
+        ShapeRelation relation = randomFrom(ShapeRelation.values());
+        DateTime from = DateTime.now();
+        DateTime to = DateTime.now().plusDays(DISTANCE);
+
+        // Test default date format
+        DateMathParser defaultParser = new DateMathParser(DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER);
+        assertEquals(
+            RangeType.DATE.rangeQuery(FIELDNAME, true, from, to, true,
+                true, relation, null, defaultParser, context),
+            fieldType.rangeQuery(from, to, true, true, relation, null, null, context));
```

Comment:
I'm not sure what is tested here, can you explain? 
As far as I can see, `fieldType.rangeQuery` only redirects to `DATE#rangeQuery`, so other than checking if the parser gets picked up, this test is not doing much in my opinion, but I might miss something.
I'd prefer it it would check that the from/to values are parsed correctly using the parser supplied by `fieldType.setDateTimeFormatter`. For that I think the from/to values should be String representations of dates that DEFAULT_DATE_TIME_FORMATTER would reject otherwise. I'd like to see a test that checks that using the DEFAULT_DATE_TIME_FORMATTER we get an error (expectThrows) and with the right parser the query contains the expected min/max values. Accessing the min/max values of the Lucene query might be a bit tricky on first inspection of the code, but I think most shape relations should produce LongRange queries which Query can be cast to to use the getters.